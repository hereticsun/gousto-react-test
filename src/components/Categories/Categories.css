.categories {

}

.categories__heading {
  
}

.categories__list {
  display: flex;
  flex-flow: row wrap;
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.categories__category {
  display: flex;
  flex: 0 0 auto;
  align-items: center;
  justify-content: center;
  margin: 8px;
  padding: 8px 16px;
  border-radius: 3px;
  color: #fff;
  background-color: #7290ff;
}
