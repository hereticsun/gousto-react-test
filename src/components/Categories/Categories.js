import React, { Component } from 'react'
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { fetchCategories } from '../../actions/categories';
import './Categories.css';

export class Categories extends Component {
  componentDidMount() {
    this.props.fetchCategories();
  }

  renderList() {
    return this.props.categories.map((category) => {
      return (
            <li
                key={category.id}
                className="categories__category"
            >
                {category.title}
            </li>
        )
    })
  }

  render() {
    return (
      <section className="categories">
        <h2 className="categories__heading">Categories:</h2>
        {this.props.categories.length > 0 ? (
          <ul className="categories__list">
              {this.renderList()}
          </ul>
        ) : null}
      </section>  
      )
  }
}

function mapStateToProps(state) {
  return {
      categories: state.categories
  };
}

Categories.propTypes = {
  categories: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      box_limit: PropTypes.number,
      is_default: PropTypes.bool,
      recently_added: PropTypes.bool,
      hidden: PropTypes.bool,
      pivot: PropTypes.shape({
        created_at: PropTypes.string,
      }),
    })),
};

export default connect(mapStateToProps, { fetchCategories })(Categories);
