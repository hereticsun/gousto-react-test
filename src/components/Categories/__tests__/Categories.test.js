import React from 'react';
import { shallow } from 'enzyme';

import {Categories} from '../Categories';

describe(`<Categories />`, () => {
  describe('Given that I am a user', () => {
    describe('when I land on the main page', () => {
      const props = {
        categories: [
          {
            id: '1',
            title: 'Drinks Cabinet',
          },
          {
            id: '2',
            title: 'Kitchenware',
          },
        ],
        fetchCategories: jest.fn()
      };
      const element = shallow(<Categories {...props} />);

      it('displays the categories of products', () => {
        expect(element).toMatchSnapshot();
        expect(element.find('li')).toHaveLength(props.categories.length);
      });
    });
  });
});
