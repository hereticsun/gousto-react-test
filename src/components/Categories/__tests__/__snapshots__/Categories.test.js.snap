// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> Given that I am a user when I land on the main page displays the categories of products 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Categories
    categories={
      Array [
        Object {
          "id": "1",
          "title": "Drinks Cabinet",
        },
        Object {
          "id": "2",
          "title": "Kitchenware",
        },
      ]
    }
    fetchCategories={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="categories__heading"
        >
          Categories:
        </h2>,
        <ul
          className="categories__list"
        >
          <li
            className="categories__category"
          >
            Drinks Cabinet
          </li>
          <li
            className="categories__category"
          >
            Kitchenware
          </li>
        </ul>,
      ],
      "className": "categories",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Categories:",
          "className": "categories__heading",
        },
        "ref": null,
        "rendered": "Categories:",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="categories__category"
            >
              Drinks Cabinet
            </li>,
            <li
              className="categories__category"
            >
              Kitchenware
            </li>,
          ],
          "className": "categories__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": "Drinks Cabinet",
              "className": "categories__category",
            },
            "ref": null,
            "rendered": "Drinks Cabinet",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": "Kitchenware",
              "className": "categories__category",
            },
            "ref": null,
            "rendered": "Kitchenware",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="categories__heading"
          >
            Categories:
          </h2>,
          <ul
            className="categories__list"
          >
            <li
              className="categories__category"
            >
              Drinks Cabinet
            </li>
            <li
              className="categories__category"
            >
              Kitchenware
            </li>
          </ul>,
        ],
        "className": "categories",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Categories:",
            "className": "categories__heading",
          },
          "ref": null,
          "rendered": "Categories:",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="categories__category"
              >
                Drinks Cabinet
              </li>,
              <li
                className="categories__category"
              >
                Kitchenware
              </li>,
            ],
            "className": "categories__list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": "Drinks Cabinet",
                "className": "categories__category",
              },
              "ref": null,
              "rendered": "Drinks Cabinet",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "host",
              "props": Object {
                "children": "Kitchenware",
                "className": "categories__category",
              },
              "ref": null,
              "rendered": "Kitchenware",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
